addons/web/static/src/js/fields/signature.js

```diff
@@ -230,19 +231,13 @@ var FieldBinarySignature = AbstractFieldBinary.extend({
                 noInputName: true,
             };
 
-            if (this.nodeOptions.full_name) {
-                var $search = $("[name='" + this.nodeOptions.full_name + "']");
-                // If full_name correspond to field in read-only mode
-                var signName = $($search).text();
-
-                // If full_name correspond to field to fill
-                if (!signName && $($search)[0]) {
-                    signName = $($search)[0].value;
-                }
-                if (!signName) {
-                    // If full_name is in the recordData. Last option because this value might not be the last value.
-                    signName = this.recordData[this.nodeOptions.full_name] || '';
-                }
+            if (this.nodeOptions.full_name && this.fields[this.nodeOptions.full_name]) {
```

Comment:
I would simplify the condition here, because it makes no sense to specify a `full_name` option with a field that does not exist. Let's assume that the view developer knows what he is doing.

`if (this.nodeOptions.full_name)`